project-base/storefront/components/Basic/GoogleMap/GoogleMap.tsx

```diff
@@ -0,0 +1,142 @@
+import GoogleMapReact from 'google-map-react';
+import getConfig from 'next/config';
+import { useEffect, useMemo, useState } from 'react';
+import useSupercluster from 'use-supercluster';
+
+type GoogleMapMarker = {
+    identifier?: string;
+    uuid?: string;
+    locationLatitude: string | null;
+    locationLongitude: string | null;
+};
+
+type GoogleMapProps = {
+    lat?: string | null;
+    lng?: string | null;
+    zoom?: number | null;
+    markers?: GoogleMapMarker[];
+    activeMarkerHandler?: (id: string) => void;
+    isDetail?: boolean;
+    closeMarkers?: boolean;
+};
+
+const TEST_IDENTIFIER = 'basic-googlemap';
+
+const ClusterMarker: FC<{
+    lat: number;
+    lng: number;
+}> = ({ children }) => (
+    <div className="absolute -translate-x-1/2 -translate-y-1/2">
+        <div className="flex justify-center items-center w-12 h-12 bg-white border-8 border-transparent rounded-full font-bold">
+            {children}
+        </div>
+    </div>
+);
+
+export const GoogleMap: FC<GoogleMapProps> = ({
+    lat,
+    lng,
+    zoom: zoomProps,
+    markers,
+    activeMarkerHandler,
+    isDetail,
+    closeMarkers,
+}) => {
+    const { publicRuntimeConfig } = getConfig();
+    const { mapSetting } = useDomainConfig();
+    const defaultLat = lat === null || lat === undefined ? mapSetting.latitude : parseFloat(lat);
+    const defaultLng = lng === null || lng === undefined ? mapSetting.longitude : parseFloat(lng);
+    const defaultZoom = zoomProps === null || zoomProps === undefined ? mapSetting.zoom : zoomProps;
+    const [activeMarkerId, setActiveMarkerId] = useState('');
+    const [zoom, setZoom] = useState(defaultZoom);
+    const [bounds, setBounds] = useState<number[]>([]);
+
+    const points = useMemo(
+        () =>
+            markers?.map((marker) => ({
+                type: 'Feature',
+                properties: {
+                    cluster: false,
+                    markerId: marker.identifier ?? marker.uuid,
+                },
+                geometry: {
+                    type: 'Point',
+                    coordinates: [
+                        marker.locationLongitude !== null ? parseFloat(marker.locationLongitude) : null,
+                        marker.locationLatitude !== null ? parseFloat(marker.locationLatitude) : null,
+                    ],
+                },
+            })) ?? [],
+        [markers],
+    );
+
+    const { clusters } = useSupercluster({
+        points,
+        zoom,
+        bounds,
+        options: { radius: 75, maxZoom: 20 },
+    });
+
+    const markerClickHandler = (id: string) => {
+        if (isDetail) {
+            return;
+        }
+        const newActiveMarkerId = activeMarkerId === id ? '' : id;
+        setActiveMarkerId(newActiveMarkerId);
+        activeMarkerHandler?.(newActiveMarkerId);
+    };
+
+    useEffect(() => {
+        if (!closeMarkers) {
+            return;
+        }
+        setActiveMarkerId('');
```

Comment:
Early returns from useEffects are apparently an anti-pattern (don't ask me why, this is what we agreed on quite recently 😆 ). Here, I would suggest this change:

```suggestion
        if !closeMarkers) {
            setActiveMarkerId('');
        }
```

Reply:
Ok I don't quite agree on this one but I will respect it :-D